### PR TITLE
Add: Implement `get_byte_order` and `dec2str` NASL functions for Rust implementation

### PR DESCRIPTION
**Pull request by jafarlihi**
**What**:

Implement NASL `get_byte_order` and `dec2str` functions for Rust implementation.

**Why**:

It is a built-in NASL function.

**How**:

Small unit tests.

**Checklist**:

- [x] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: rust/nasl-interpreter/src/built_in_functions/misc.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+/// NASL function to get host byte order
+pub fn get_byte_order(_: &str, _: &dyn Sink, _: &Register) -> Result<NaslValue, FunctionError> {
+    Ok(NaslValue::Boolean(cfg!(target_endian = "little")))
+}
+
+/// NASL function to convert given number to string
+pub fn dec2str(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    match register.named("num") {
+        Some(ContextType::Value(NaslValue::Number(x))) => Ok(NaslValue::String(x.to_string())),
+        _ => Err(FunctionError::new(
```
Comment (nichtsfrei) [inline rust/nasl-interpreter/src/built_in_functions/misc.rs:48]: I think it needs to be changed to
```
x => Err(FunctionError::new("dec2str", ("1", "numeric", x).into())),
```

File: rust/nasl-interpreter/src/built_in_functions/misc.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+use std::{
+    fs::File,
+    io::{self, Read},
+};
+
+use sink::Sink;
+
+use crate::{error::FunctionError, ContextType, NaslFunction, NaslValue, Register};
+
+impl From<io::Error> for FunctionError {
```
Comment (nichtsfrei) [inline rust/nasl-interpreter/src/built_in_functions/misc.rs:16]: That is moved to error.rs and be deleted.

File: rust/nasl-interpreter/src/built_in_functions/misc.rs (hunk 1 above)
Comment (jafarlihi) [reply rust/nasl-interpreter/src/built_in_functions/misc.rs]: Done

File: rust/nasl-interpreter/src/built_in_functions/misc.rs (hunk 2 above)
Comment (jafarlihi) [reply rust/nasl-interpreter/src/built_in_functions/misc.rs]: Done

**Review by nichtsfrei — APPROVED** [review]

**Review by Kraemii — APPROVED** [review]

**Pull request merged**